dataset/table.py

```diff
@@ -1,78 +1,88 @@
+            self.df = pd.read_csv(fpath, dtype=str, na_values=na_values, encoding='utf-8')
+            # Normalize to lower strings and strip whitespaces.
+            for attr in self.df.columns.values:
+                if attr not in exclude_attr_cols:
+                    self.df[attr] = self.df[attr].str.strip().str.lower()
         elif src == Source.DF:
-            if len(args) != 1:
-                raise Exception("ERROR while loading table. Dataframe expected. Please provide <dataframe>.")
-            else:
-                self.df = args[0]
+            if df is None:
+                raise Exception("ERROR while loading table. Dataframe expected. Please provide <df> param.")
+            self.df = df
         elif src == Source.DB:
-            if len(args) != 1:
+            if db_conn is None:
                 raise Exception("ERROR while loading table. DB connection expected. Please provide <db_conn>")
-            else:
-                db_conn = args[0]
-                self.df = pd.read_sql_table(name, db_conn)
+            self.df = pd.read_sql_table(name, db_conn)
         elif src == Source.SQL:
-            if len(args) != 2:
-                raise Exception("ERROR while loading table. SQL Query and DB engine expected. Please provide <query> and <db_engine>.")
-            else:
-                tab_query = args[0]
-                dbengine = args[1]
-                dbengine.create_db_table_from_query(self.name, tab_query)
-                self.df = pd.read_sql_table(name, dbengine.conn)
+            if table_query is None or db_engine is None:
+                raise Exception("ERROR while loading table. SQL Query and DB connection expected. Please provide <table_query> and <db_engine>.")
+            db_engine.create_db_table_from_query(self.name, table_query)
+            self.df = pd.read_sql_table(name, db_engine.conn)
 
-    def store_to_db(self, con, if_exists='replace', index=False, index_label=None):
+    def store_to_db(self, db_conn, if_exists='replace', index=False, index_label=None):
         # TODO: This version supports single session, single worker.
-        self.df.to_sql(self.name, con, if_exists=if_exists, index=index, index_label=index_label)
+        self.df.to_sql(self.name, db_conn, if_exists=if_exists, index=index, index_label=index_label)
 
     def get_attributes(self):
-        if not self.df.empty:
-            return list(self.df.columns.values)
-        else:
-            raise Exception("Empty Dataframe associated with table "+self.name+". Cannot return attributes.")
+        """
+        get_attributes returns the columns that are trainable/learnable attributes
+        (i.e. exclude meta-columns like _tid_).
+        """
+        if self.df.empty:
+            raise Exception("Empty Dataframe associated with table {name}. Cannot return attributes.".format(
+                name=self.name))
+        return list(col for col in self.df.columns if col not in self.exclude_attr_cols)
```

Comment:
Could just use list comprehension directly if a list is being returned.
```suggestion
        return [col for col in self.df.columns if col not in self.exclude_attr_cols]
```